main/client/src/mill/main/client/MillClientMain.java

```diff
@@ -19,32 +19,43 @@
 	public static final int ExitServerCodeWhenVersionMismatch() { return 101; }
 
     static void initServer(String lockBase, boolean setJnaNoSys) throws IOException,URISyntaxException{
-        String[] selfJars = System.getProperty("MILL_CLASSPATH").split(",");
-
-        List<String> l = new ArrayList<>();
+        
+        String selfJars = "";
         List<String> vmOptions = new ArrayList<>();
-        l.add(System.getProperty("java.home") + File.separator + "bin" + File.separator + "java");
-        final Properties props = System.getProperties();
-        for(final String k: props.stringPropertyNames()){
+        String millOptionsPath = System.getProperty("MILL_OPTIONS_PATH");
+        if(millOptionsPath != null) {
+            // read MILL_CLASSPATH from file MILL_OPTIONS_PATH
+            Properties millProps = new Properties();
```

Comment:
On Windows we read the classpath from the `MILL_OPTIONS_PATH` file instead of env var..